src/azure-cli/azure/cli/command_modules/acr/connected_registry.py

```diff
@@ -0,0 +1,494 @@
+    repository_actions_list = [[repo] + REPO_SCOPES_BY_MODE[mode] for repo in repositories]
+    gateway_actions_list = [[connected_registry_name] + DEFAULT_GATEWAY_SCOPE]
+    try:
+        message = "Created by connected registry sync token: {}"
+        sync_scope_map_name = SYNC_SCOPE_MAP_NAME.format(connected_registry_name)
+        logger.warning("If sync scope map '%s' already exist properties will be overrided", sync_scope_map_name)
+        sync_scope_map = create_default_scope_map(cmd, resource_group_name, registry_name, sync_scope_map_name,
+                                                  repository_actions_list, gateway_actions_list,
+                                                  scope_map_description=message.format(connected_registry_name),
+                                                  force=True)
+
+        sync_token_name = SYNC_TOKEN_NAME.format(connected_registry_name)
+        logger.warning("If sync token '%s' already exist properties will be overrided", sync_token_name)
+        Token = cmd.get_models('Token')
+        poller = token_client.create(
+            resource_group_name,
+            registry_name,
+            sync_token_name,
+            Token(
+                scope_map_id=sync_scope_map.id,
+                status="enabled"
+            )
+        )
+
+        token = LongRunningOperation(cmd.cli_ctx)(poller)
+        return token.id
+    except ValidationError as e:
+        raise CLIError(e)
+
+
+def _get_descendancy(family_tree, parent_id):
+    childs = family_tree[parent_id]['childs']
+    result = []
+    for child_id in childs:
+        result = [family_tree[child_id]["registry"]]
+        descendancy = _get_descendancy(family_tree, child_id)
+        if descendancy:
+            result.extend(descendancy)
+    return result
+
+
+#TODO delete if not used
+def _verify_parent_permissions(cmd,
+                               registry_name,
+                               connected_registry_name,
+                               parent):
+    sync_token = get_token_from_id(cmd, parent.parent.sync_properties.token_id)
+    parent_sync_scope_map = get_scope_map_from_id(cmd, sync_token.scope_map_id)
+    parent_sync_actions = parent_sync_scope_map.actions
+    gateway_list = [[connected_registry_name] + DEFAULT_GATEWAY_SCOPE]
+    gateway_actions = parse_actions_from_gateways(gateway_list)
+
+    # TODO should I loop through all parents?
+    if not all(action in parent_sync_actions  for action in gateway_actions):
+        raise CLIError(
+            "Parent '{}' sync token '{}'. ".format(parent.name, sync_token.name) +
+            "Ensure the sync token has the following permissions: \n\t{}\n".format(gateway_actions) +
+            "Run the command: az acr scope-map update -n {} -r {} --add-gateway {}".format(
+                connected_registry_name, registry_name, gateway_list
+            ))
+
+
+#TODO delete if not used
```

Comment:
Is this still needed?

Reply:
I'll delete all those methods before merging. But there has been a lot of back and forward on the CLI behavior. So I left this methods just in case. We needed them again. But that doesn't seems to be the case.